src/main/scala/thurloe/service/ThurloeService.scala

```diff
@@ -24,11 +25,16 @@ trait ThurloeService extends HttpService {
 
   val getRoute = path(ThurloePrefix / Segment / Segment) { (userId, key) =>
     get {
-      onComplete(dataAccess.lookup(userId, key)) {
-        case Success(keyValuePairWithId) =>
+      val query: Future[Seq[UserKeyValuePair]] = if ("*".equals(userId)) { dataAccess.lookupByKey(key) } else { dataAccess.lookup(userId, key) map { Seq(_) } }
```

Comment:
This feels strange to me too, but if you do stick with this scheme it should be documented in the YAML as it's not obvious how this works.